token/program-2022/tests/transfer_fee.rs

```diff
@@ -993,3 +1007,215 @@ async fn no_fees_from_self_transfer() {
+        alice,
+        &alice.pubkey(),
+        amount,
+        *decimals,
+    )
+    .await;
+    context
+        .init_token_with_mint(vec![ExtensionInitializationParams::TransferFeeConfig {
+            transfer_fee_config_authority: transfer_fee_config_authority.pubkey().into(),
+            withdraw_withheld_authority: withdraw_withheld_authority.pubkey().into(),
+            transfer_fee_basis_points,
+            maximum_fee,
+        }])
+        .await
+        .unwrap();
+    let TokenContext { token, .. } = context.token_context.unwrap();
+    token
+        .harvest_withheld_tokens_to_mint(&[account])
+        .await
+        .unwrap();
+    let state = token.get_mint_info().await.unwrap();
+    let extension = state.get_extension::<TransferFeeConfig>().unwrap();
+    assert_eq!(extension.withheld_amount, 0.into());
+}
+
+#[tokio::test]
+async fn max_harvest_withheld_tokens_to_mint() {
+    let TransferFeeConfigWithKeypairs {
+        transfer_fee_config_authority,
+        withdraw_withheld_authority,
+        transfer_fee_config,
+        ..
+    } = test_transfer_fee_config_with_keypairs();
+    let mut context = TestContext::new().await;
+    let transfer_fee_basis_points = u16::from(
+        transfer_fee_config
+            .newer_transfer_fee
+            .transfer_fee_basis_points,
+    );
+    let maximum_fee = u64::from(transfer_fee_config.newer_transfer_fee.maximum_fee);
+    context
+        .init_token_with_mint(vec![ExtensionInitializationParams::TransferFeeConfig {
+            transfer_fee_config_authority: transfer_fee_config_authority.pubkey().into(),
+            withdraw_withheld_authority: withdraw_withheld_authority.pubkey().into(),
+            transfer_fee_basis_points,
+            maximum_fee,
+        }])
+        .await
+        .unwrap();
+    let TokenContext {
+        decimals,
+        mint_authority,
+        token,
+        alice,
+        ..
+    } = context.token_context.unwrap();
+
+    let alice_account = Keypair::new();
+    let alice_account = token
+        .create_auxiliary_token_account(&alice_account, &alice.pubkey())
+        .await
+        .unwrap();
+
+    // mint a lot of tokens
+    let amount = maximum_fee;
+    let alice_amount = amount * 100;
+    token
+        .mint_to(&alice_account, &mint_authority, alice_amount)
+        .await
+        .unwrap();
+    // harvest from max accounts, which is around 35, AKA 34 accounts + 1 mint
+    // see https://docs.solana.com/proposals/transactions-v2#problem
+    let mut accounts = vec![];
+    let max_accounts = 34;
+    for _ in 0..max_accounts {
+        let account = create_and_transfer_to_account(
+            &token,
+            &alice_account,
+            &alice,
+            &alice.pubkey(),
+            amount,
+            decimals,
+        )
+        .await;
+        accounts.push(account);
+    }
+    let accumulated_fees =
+        max_accounts * transfer_fee_config.calculate_epoch_fee(0, amount).unwrap();
+    token
```

Comment:
```suggestion
    let accounts: Vec<_> = accounts.iter().collect();
    token
```
If you oblige my `&Pubkey` request